data_structures/binary_heap.ts

```diff
@@ -71,7 +71,7 @@ export class BinaryHeap<T> implements Iterable<T> {
   constructor(compare: (a: T, b: T) => number = descend) {
     if (typeof compare !== "function") {
       throw new TypeError(
-        "compare must be a function, did you mean to use BinaryHeap.from?",
+        "The 'compare' parameter must be a function, did you mean to use BinaryHeap.from?",
```

Comment:
```suggestion
        "The 'compare' parameter must be a function. Did you mean to use BinaryHeap.from?",
```

Reply:
The rules we have indicate that a period should not be used within an error message. The idea is that we don't want several sentences, which is why I've been trying to use a single sentence (possibly with a colon). Do we want to change this rule?

Reply:
What do Web APIs, V8, Deno CLI do?

Reply:
I think V8 uses commas: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_define_property_object_not_extensible

Reply:
Just found an example with colon: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Regex_nothing_to_repeat

Reply:
I think the original rule said either comma `,` or colon `:`, but not period. I like that rule as there are often times a comma makes sense, but we don't want several sentences (that's why no period).

I've updated a number of these error messages, and without the use of a comma, I think they get very hard to read.

Reply:
```suggestion
        "Cannot construct a BinaryHeap: the 'compare' parameter is not a function, did you mean to call BinaryHeap.from?",
```